noir-projects/aztec-nr/aztec/src/state_vars/shared_mutable.nr

```diff
@@ -24,21 +23,11 @@ pub struct SharedMutable<T, let INITIAL_DELAY: u32, Context> {
     storage_slot: Field,
 }
 
-// Separators separating storage slot of different values within the same state variable
-global VALUE_CHANGE_SEPARATOR: u32 = 0;
-global DELAY_CHANGE_SEPARATOR: u32 = 1;
-global HASH_SEPARATOR: u32 = 2;
-
-// This will make the Aztec macros require that T implements the Serialize<N> trait, and allocate N storage slots to
-// this state variable. This is incorrect, since what we actually store is:
-//  - a ScheduledValueChange<T>, which requires 1 + 2 * M storage slots, where M is the serialization length of T
-//  - a ScheduledDelayChange, which requires another storage slot
-//
-// TODO https://github.com/AztecProtocol/aztec-packages/issues/5736: change the storage allocation scheme so that we
-// can actually use it here
+// This will make the Aztec macros require that T implements the Packable<N> and Eq traits, and allocate `N` storage
+// slots to this state variable.
 impl<T, let INITIAL_DELAY: u32, Context, let N: u32> Storage<N> for SharedMutable<T, INITIAL_DELAY, Context>
 where
-    T: Packable<N>,
+    WithHash<SharedMutableValues<T, INITIAL_DELAY>, _>: Packable<N>,
```

Comment:
I'd consider calling this `N` something else, since in the other impls `N` is the packing len of T.

Reply:
Makes sense. Renamed in https://github.com/AztecProtocol/aztec-packages/pull/11716/commits/251e801719f6e17797e39852f788c694dfaa7413